server/src/main/java/org/tctalent/server/service/db/impl/UserServiceImpl.java

```diff
@@ -527,6 +527,15 @@ public User getLoggedInUser() {
         return user;
     }
 
+    @Override
+    public User getUserByEmailVerificationToken(String token) throws NoSuchObjectException {
+        User user = userRepository.findByEmailVerificationToken(token);
+        if (user == null) {
+            throw new NoSuchObjectException("Invalid or expired token");
```

Comment:
Only detects invalid tokens.

```suggestion
            throw new NoSuchObjectException("Invalid token");
```